bgpd/bgp_updgrp.c

```diff
@@ -703,79 +712,231 @@ static int update_group_show_walkcb(struct update_group *updgrp, void *arg)
 
 	vty = ctx->vty;
 
-	vty_out(vty, "Update-group %" PRIu64 ":\n", updgrp->id);
-	vty_out(vty, "  Created: %s", timestamp_string(updgrp->uptime));
+	if (ctx->uj) {
+		json_updgrp = json_object_new_object();
+		/* Display json o/p */
+		tbuf = monotime(NULL);
+		tbuf -= updgrp->uptime;
+		epoch_tbuf = time(NULL) - tbuf;
+		json_time = json_object_new_object();
+		json_object_int_add(json_time, "epoch", epoch_tbuf);
+		json_object_string_add(json_time, "epochString",
+				       ctime(&epoch_tbuf));
+		json_object_object_add(json_updgrp, "groupCreateTime",
+				       json_time);
+		json_object_string_add(json_updgrp, "afi",
+				       afi2str(updgrp->afi));
+		json_object_string_add(json_updgrp, "safi",
+				       safi2str(updgrp->safi));
+	} else {
+		vty_out(vty, "Update-group %" PRIu64 ":\n", updgrp->id);
+		vty_out(vty, "  Created: %s", timestamp_string(updgrp->uptime));
+	}
+
 	filter = &updgrp->conf->filter[updgrp->afi][updgrp->safi];
-	if (filter->map[RMAP_OUT].name)
-		vty_out(vty, "  Outgoing route map: %s\n",
-			filter->map[RMAP_OUT].name);
-	vty_out(vty, "  MRAI value (seconds): %d\n", updgrp->conf->v_routeadv);
-	if (updgrp->conf->change_local_as)
-		vty_out(vty, "  Local AS %u%s%s\n",
-			updgrp->conf->change_local_as,
-			CHECK_FLAG(updgrp->conf->flags,
-				   PEER_FLAG_LOCAL_AS_NO_PREPEND)
-				? " no-prepend"
-				: "",
-			CHECK_FLAG(updgrp->conf->flags,
-				   PEER_FLAG_LOCAL_AS_REPLACE_AS)
-				? " replace-as"
-				: "");
+	if (filter->map[RMAP_OUT].name) {
+		if (ctx->uj)
+			json_object_string_add(json_updgrp, "outRouteMap",
+					       filter->map[RMAP_OUT].name);
+		else
+			vty_out(vty, "  Outgoing route map: %s\n",
+				filter->map[RMAP_OUT].name);
+	}
 
+	if (ctx->uj)
+		json_object_int_add(json_updgrp, "minRouteAdvInt",
+				    updgrp->conf->v_routeadv);
+	else
+		vty_out(vty, "  MRAI value (seconds): %d\n",
+			updgrp->conf->v_routeadv);
+
+	if (updgrp->conf->change_local_as) {
+		if (ctx->uj) {
+			json_object_int_add(json_updgrp, "localAs",
+					    updgrp->conf->change_local_as);
+			json_object_boolean_add(
+				json_updgrp, "noPrependFlag",
```

Comment:
nit: I'm just about consistency that noPrependFlag (flag suffix), while replaceLocalAs (no flag suffix).

Reply:
Thanks @ton31337  for spotting.
I've corrected all flags